File: mobile/src/utils/categories.ts
import type { TFunction } from 'i18next';

import i18n from '@/src/i18n';

export const CATEGORY_EMOJI: Record<string, string> = {
  museum: '🏛️',
  landmark: '🗿',
  'cultural-spot': '🎭',
  beach: '🏖️',
  'walking-area': '🚶',
  cafe: '☕',
  restaurant: '🍽️',
  viewpoint: '🌅',
  nature: '🌿',
  'shopping-area': '🛍️',
  lodging: '🏨',
  'square-street': '🏙️',
};

const CATEGORY_LABEL_KEYS: Record<string, string> = {
  landmark: 'categories.landmark',
  museum: 'categories.museum',
  'cultural-spot': 'categories.culturalSpot',
  'square-street': 'categories.squareStreet',
  beach: 'categories.beach',
  'walking-area': 'categories.walkingArea',
  cafe: 'categories.cafe',
  restaurant: 'categories.restaurant',
  viewpoint: 'categories.viewpoint',
  'shopping-area': 'categories.shoppingArea',
  lodging: 'categories.lodging',
  nature: 'categories.nature',
};

function defaultT(key: string, options?: Record<string, unknown>) {
  return i18n.t(key, options as any) as string;
}

export function categoryEmoji(c: string): string {
  return CATEGORY_EMOJI[c] ?? '📍';
}

export function formatCategory(c: string, t: TFunction = defaultT as TFunction): string {
  const key = CATEGORY_LABEL_KEYS[c];
  if (key) return t(key) as string;
  return c.split('-').map((w) => w[0].toUpperCase() + w.slice(1)).join(' ');
}

File: native/Piri/Sources/Utils/Categories.swift
import Foundation

/// Port of `mobile/src/utils/categories.ts`.
enum Categories {
    static let icon: [String: String] = [
        "museum": "building.columns.fill",
        "landmark": "mappin.and.ellipse",
        "cultural-spot": "theatermasks.fill",
        "beach": "beach.umbrella.fill",
        "walking-area": "figure.walk",
        "cafe": "cup.and.saucer.fill",
        "restaurant": "fork.knife",
        "viewpoint": "sun.horizon.fill",
        "nature": "leaf.fill",
        "shopping-area": "bag.fill",
        "lodging": "bed.double.fill",
        "square-street": "building.2.fill",
    ]

    private static let labelKeys: [String: String] = [
        "landmark": "categories.landmark",
        "museum": "categories.museum",
        "cultural-spot": "categories.culturalSpot",
        "square-street": "categories.squareStreet",
        "beach": "categories.beach",
        "walking-area": "categories.walkingArea",
        "cafe": "categories.cafe",
        "restaurant": "categories.restaurant",
        "viewpoint": "categories.viewpoint",
        "shopping-area": "categories.shoppingArea",
        "lodging": "categories.lodging",
        "nature": "categories.nature",
    ]

    static func icon(for category: String) -> String {
        icon[category] ?? "mappin.circle.fill"
    }

    static func label(for category: String) -> String {
        guard let key = labelKeys[category] else {
            return category.split(separator: "-").map { $0.prefix(1).uppercased() + $0.dropFirst() }.joined(separator: " ")
        }
        return String(localized: String.LocalizationValue(key))
    }
}
